examples/tutorials/asr_inference_with_ctc_decoder_tutorial.py

```diff
@@ -222,28 +228,215 @@ def forward(self, emission: torch.Tensor) -> str:
+#
+# This parameter indicates the number of best Hypothesis to return, which
+# is a property that is not possible with the greedy decoder. For
+# instance, by setting ``nbest=3`` when constructing the beam search
+# decoder earlier, we can now access the hypotheses with the top 3 scores.
+#
+
+for i in range(3):
+    transcript = " ".join(beam_search_result[0][i].words).strip()
+    score = beam_search_result[0][i].score
+    print(f"{transcript} (score: {score})")
+
+
+######################################################################
+# beam size
+# ~~~~~~~~~
+#
+# The ``beam_size`` parameter determines the maximum number of best
+# hypotheses to hold after each decoding step. Using larger beam sizes
+# allows for exploring a larger range of possible hypotheses which can
+# produce hypotheses with higher scores, but it is computationally more
+# expensive and does not provide additional gains beyond a certain point.
+#
+# In the example below, we see improvement in decoding quality as we
+# increase beam size from 1 to 5 to 50, but notice how using a beam size
+# of 500 provides the same output as beam size 50 while increase the
+# computation time.
+#
+
+beam_sizes = [1, 5, 50, 500]
+
+for beam_size in beam_sizes:
+    beam_search_decoder = kenlm_lexicon_decoder(
+        lexicon=lexicon_file,
+        tokens=tokens,
+        kenlm=kenlm_file,
+        beam_size=beam_size,
+        lm_weight=LM_WEIGHT,
+        word_score=WORD_SCORE,
+    )
+
+    transcript, decode_time = get_transcript_and_time(beam_search_decoder, emission)
+    print(f"beam size {beam_size:<3}: {transcript} ({decode_time:.4f} secs)")
+
+
+######################################################################
+# beam size token
+# ~~~~~~~~~~~~~~~
+#
+# The ``beam_size_token`` parameter corresponds to the number of tokens to
+# consider for expanding each hypothesis at the decoding step. Exploring a
+# larger number of next possible tokens increases the range of potential
+# hypotheses at the cost of computation. This parameter is especially
+# helpful for lexicon-free decoders, where the number of possible next
+# tokens is not already partially constrained by the lexicon.
+#
+
+num_tokens = len(tokens)
+beam_size_tokens = [1, 5, 10, num_tokens]
+
+for beam_size_token in beam_size_tokens:
+    beam_search_decoder = kenlm_lexicon_decoder(
+        lexicon=lexicon_file,
+        tokens=tokens,
+        kenlm=kenlm_file,
+        beam_size_token=beam_size_token,
+        lm_weight=LM_WEIGHT,
+        word_score=WORD_SCORE,
+    )
+
+    transcript, decode_time = get_transcript_and_time(beam_search_decoder, emission)
+    print(f"beam size token {beam_size_token:<3}: {transcript} ({decode_time:.4f} secs)")
+
+
+######################################################################
+# beam threshold
+# ~~~~~~~~~~~~~~
+#
+# The ``beam_threshold`` parameter is used to prune the stored hypotheses
+# set at each decoding step, removing hypotheses whose scores are greater
```

Comment:
Isn't the higher the better?
https://514632-90321822-gh.circle-artifacts.com/0/docs/tutorials/asr_inference_with_ctc_decoder_tutorial.html#beam-size-token

Also, can you also show scores for each example after `nbest`? I was trying to check the score of other examples.